usaspending_api/search/tests/unit/filters/naics.py

```diff
@@ -0,0 +1,37 @@
+from usaspending_api.search.elasticsearch.filters.naics import NaicsCodes
```

Comment:
👍 this is a good start. i was really hoping to see this, because that code is fraught with subtle assumptions about what is and is not possible, with zero documentation. 

However, I think you ought to add more of these test cases. 
e.g. specifically, add some tests with _many_ elements in the arrays, that have overlapping prefixes, and just many in general, and assert that the result is what was expected. 
Also if order has any bearing, try in the default order, then reverse the arrays and try again.

Reply:
Can you give an example for "overlapping prefixes"?

Reply:
There is overlap among these 3
```
"naics_codes": {
    "require": ["11", "112233"],
    "exclude": ["1122"]
}
```

And these three
```
"naics_codes": {
    "require": ["1122"],
    "exclude": ["11", "112233"]
}
```

And these three
```
"naics_codes": {
    "require": ["112233"],
    "exclude": ["11", "1122"]
}
```

And ...
```
"naics_codes": {
    "require": ["11", "112233"],
    "exclude": ["112233"]
}
```
And
```
"naics_codes": {
    "require": ["112233", "112233"],
    "exclude": ["112233", "112233"]
}
```
And
```
"naics_codes": {
    "require": ["11", "112233"],
    "exclude": ["1122"]
}
```
And
```
"naics_codes": {
    "require": ["112233", "112233"],
    "exclude": ["112233", "1122", "112233"]
}
```

While you're at it, it would also be good to add a unit test to verify logic when an empty string, 1-digit string, 3-digit string, or 5-digit string or 7+ digit string is passed.

Reply:
I was thinking to limit setup time on each test, you could have multiple asserts in one test, modifying the data for each part of the test. That way you could pack a bunch of different `require`/`exclude` combos into 1 test. But these 3 are fine.